src/cdk-experimental/menu/menu-item-trigger.ts

```diff
@@ -143,7 +148,10 @@ export class CdkMenuItemTrigger implements OnDestroy {
    * content to change dynamically and be reflected in the application.
    */
   private _getPortal() {
-    if (!this._panelContent || this._panelContent.templateRef !== this._menuPanel?._templateRef) {
+    if (
+      !this._panelContent ||
+      (this._menuPanel && this._panelContent.templateRef !== this._menuPanel._templateRef)
+    ) {
```

Comment:
```suggestion
    const hasMenuContentChanged = this._menuPanel?.templateRef !== this._panelContent?.templateRef;
    if (!this._panelContent || hasMenuContentChanged) {
```
I would split out another variable for this complex conditional to make it more readable